Avoid `catch` blocks that rethrow a caught exception wrapped inside a new instance of the same exception.

[Source](https://github.com/arturbosch/detekt/)